module/zfs/arc.c

```diff
@@ -4923,6 +4923,13 @@ arc_reclaim_thread(void *unused)
 #endif
 		mutex_exit(&arc_reclaim_lock);
 
+		/* If arc size if 0, or below c_min, there
+		 * is no need to call arc_adjust as long as
+		 * the meta usage is under the limits.
+		 */
+		if (arc_size <= MIN(arc_meta_limit, arc_c_min))
+			goto skip_adjust;
```

Comment:
I'm a little bit concerned about how to document this. It is not uncommon to see arc_c_min set high to alleviate other issues. See zfs_arc_min and zfs_arc_meta_min in zfs-module-parameters.5. Two cases come to mind:

1. It might not be possible to free memory below the MIN() without unloading the zfs module. 
2. during cases of extreme memory pressure, this might not do the right thing. 

Have you developed a test for extreme memory pressure and observed the effects with arc_c_min > the default?